Tests/ConfidenceProviderTests/ConfidenceFeatureProviderTest.swift

```diff
@@ -24,6 +25,41 @@ class ConfidenceFeatureProviderTest: XCTestCase {
         super.setUp()
     }
 
+    func testSlowFirstResolveWillbeCancelledOnSecondResolve() async throws {
+        class FakeClient: ConfidenceResolveClient {
+            var callCount = 0
+            var returnCount = 0
+
+            func resolve(ctx: ConfidenceStruct) async throws -> ResolvesResult {
+                callCount += 1
+                if callCount == 1 {
+                    // wait 2 seconds
+                    try await Task.sleep(nanoseconds: UInt64(1 * Double(NSEC_PER_MSEC)))
+                    returnCount += 1
+                } else {
+                    returnCount += 1
+                }
+
+                return .init(resolvedValues: [], resolveToken: "")
+            }
+        }
+
+        let confidence = Confidence.Builder.init(clientSecret: "").build()
+        let client = FakeClient()
+        let provider = ConfidenceFeatureProvider(confidence: confidence, session: nil, client: client)
+        let initialContext = MutableContext(targetingKey: "user1")
+            .add(key: "hello", value: Value.string("world"))
+        provider.initialize(initialContext: initialContext)
+        try await Task.sleep(nanoseconds: UInt64(2 * Double(NSEC_PER_MSEC)))
+        client.callCount = 0
+        client.returnCount = 0
+        confidence.putContext(key: "new", value: ConfidenceValue(string: "value"))
+        confidence.putContext(key: "new2", value: ConfidenceValue(string: "value2"))
+        try await Task.sleep(nanoseconds: UInt64(2 * Double(NSEC_PER_MSEC)))
+        XCTAssertEqual(2, client.callCount)
+        XCTAssertEqual(1, client.returnCount)
```

Comment:
Can we instead verify that the second putContext invocation is the one that we allow to finalize?